pkg/controller/swiftstorage/swiftstorage_controller.go

```diff
@@ -159,18 +157,86 @@ func (r *ReconcileSwiftStorage) Reconcile(request reconcile.Request) (reconcile.
 	return reconcile.Result{}, r.client.Status().Update(context.Background(), swiftStorage)
 }
 
+func (r *ReconcileSwiftStorage) ensureLocalPVs(ss *contrail.SwiftStorage) error {
+	path := ss.Spec.ServiceConfiguration.Storage.Path
+	size := ss.Spec.ServiceConfiguration.Storage.Size
+	var storage resource.Quantity
+	var err error
+	if size == "" {
+		storage = resource.MustParse("5Gi")
+	} else {
+		storage, err = resource.ParseQuantity(size)
+		if err != nil {
+			return err
+		}
+	}
+
+	if path == "" {
+		path = defaultSwiftStoragePath
+	}
+	lv, err := r.localVolumes.New(
+		fmt.Sprintf("%s-swift-data", ss.Name),
+		ss.Labels,
+		map[string]string{"node-role.kubernetes.io/master": ""},
+		path,
+		storage,
+	)
+	if err != nil {
+		return err
+	}
+	return lv.EnsureExist()
+}
+
+func (r *ReconcileSwiftStorage) ensureLabel(ss *contrail.SwiftStorage) error {
+	if len(ss.Labels) != 0 {
+		return nil
+	}
+
+	ss.Labels = contraillabel.New(contrail.SwiftStorageInstanceType, ss.Name)
+	return r.client.Update(context.Background(), ss)
```

Comment:
I'm thinking if we have this API call here, can't we set it inside createOrUpdate function?

Reply:
my bad, I see it